lib/Runtime/Types/ES5ArrayTypeHandler.cpp

```diff
@@ -271,25 +279,34 @@ namespace Js
             doneConversion = true;
 #endif
         }
+#ifdef _MSC_VER
         __finally
         {
             if (AbnormalTermination())
```

Comment:
> IMO, TryFinally wouldn't fit here as is. if (AbnormalTermination())

The [TryFinally](https://github.com/Microsoft/ChakraCore/blob/linux/lib/Common/CommonPal.h#L388) calls `finallyFunc(hasException)`. `hasException` was intended to indicate `AbnormalTermination()` here. Would it work?